### PR TITLE
[Merged by Bors] - chore(AtTopBot): split

### PR DESCRIPTION
**Pull request by urkud**
- move lemmas depending on ordered monoids to new files;
- rewrite some proofs about `Nat` to use `Nat`-specific lemmas.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
!bench

**Comment by semorrison** [comment]
bors d+

**Comment by semorrison** [comment]
Not sure why the speed center ignored you. Trying again:

!bench

**Comment by urkud** [comment]
Is it up?

**Comment by semorrison** [comment]
Looks like it is still broken, and we'll need to wait until Sebastian can restart it.

**Comment by urkud** [comment]
Let me try again
!bench

**Comment by urkud** [comment]
Merged `master` to have something to compare to.
!bench

**Comment by urkud** [comment]
Let me try again
!bench

**Comment by Kha** [comment]
!bench

**Comment by urkud** [comment]
bors merge
